core/src/main/java/org/apache/calcite/sql/validate/implicit/TypeCoercionImpl.java

```diff
@@ -94,10 +94,11 @@ public TypeCoercionImpl(RelDataTypeFactory typeFactory, SqlValidator validator)
       int columnIndex,
       RelDataType targetType) {
     final SqlKind kind = query.getKind();
+    SqlValidatorScope scope1;
```

Comment:
I don't see this variable referenced after this switch statement. It seems like this is used in order to allow a different switch statement to use the same variable. Just declare it down there. You should only use this if the variable is intended to live beyond the switch scope.

Reply:
done.